test/controllers/SecureMessagingControllerSpec.scala

```diff
@@ -137,14 +144,97 @@ class SecureMessagingControllerSpec extends ControllerSpecBase with TestRequests
           "display the 'Sorry' page to the user" in {
             secureMessagingFeatureAction.enableSecureMessagingFeature()
 
-            when(connector.retrieveConversationPartial(any(), any())(any(), any())).thenReturn(Future.failed(new Exception("Whoopse")))
+            when(connector.retrieveConversationPartial(any[String], any[String])(any[HeaderCarrier]))
+              .thenReturn(Future.failed(new Exception("Whoopse")))
 
             an[Exception] mustBe thrownBy {
-              await(controller.displayConversation(clientId, conversationId)(fakeRequest))
+              await(controller.displayConversation(clientId, conversationId)(fakeRequest.withCSRFToken))
             }
           }
         }
       }
     }
   }
+
+  "SecureMessagingController displayReplyResult is called" when {
+    val clientId = "clientId"
+    val conversationId = "conversationId"
+
+    "feature flag for SecureMessaging is disabled" should {
+
+      "throw InvalidFeatureStateException$" in {
+        secureMessagingFeatureAction.disableSecureMessagingFeature()
+
+        an[InvalidFeatureStateException] mustBe thrownBy {
+          await(controller.displayReplyResult(clientId, conversationId)(fakeRequest))
+        }
+      }
+    }
+
+    "feature flag for SecureMessaging is enabled" should {
+
+      "call secure message connector" when {
+
+        "successfully returns a ReplyResultPartial" should {
+
+          "wrap the partial in the reply_result page" in {
+            secureMessagingFeatureAction.enableSecureMessagingFeature()
+            when(connector.retrieveReplyResult(any[String], any[String])(any[HeaderCarrier]))
+              .thenReturn(Future.successful(ReplyResultPartial("")))
+
+            val result = controller.displayReplyResult(clientId, conversationId)(fakeRequest)
+
+            status(result) mustBe OK
+          }
+        }
+
+        "unsuccessfully returns a failed Future" should {
+
+          "display the 'Sorry' page to the user" in {
```

Comment:
Nit: This text is not aligned with what you assert against. It says about displaying 'Sorry' page, but assertion is against an Exception being thrown.